insights/src/main/java/org/jboss/as/insights/extension/InsightsService.java

```diff
@@ -0,0 +1,139 @@
+
+
+import static org.jboss.as.insights.logger.InsightsLogger.ROOT_LOGGER;
+
+import java.util.concurrent.ScheduledExecutorService;
+import org.jboss.as.insights.api.InsightsScheduler;
+import org.jboss.as.jdr.JdrReportCollector;
+import org.jboss.as.jdr.JdrReportService;
+import org.jboss.msc.service.Service;
+import org.jboss.msc.service.ServiceName;
+import org.jboss.msc.service.ServiceController;
+import org.jboss.msc.service.ServiceTarget;
+import org.jboss.msc.service.StartContext;
+import org.jboss.msc.service.StartException;
+import org.jboss.msc.service.StopContext;
+
+import org.jboss.msc.value.InjectedValue;
+
+/**
+ * @author <a href="mailto:jkinlaw@redhat.com">Josh Kinlaw</a>
+ */
+public class InsightsService implements Service<InsightsScheduler> {
+
+    public static final ServiceName SERVICE_NAME = ServiceName.JBOSS.append(InsightsExtension.SUBSYSTEM_NAME);
+
+    public static final String USERNAME = "username";
+    public static final String PASSWORD = "password";
+    public static final String URL = "url";
+    public static final String PROXY_USER = "proxyUser";
+    public static final String PROXY_PASSWORD = "proxyPassword";
+    public static final String PROXY_URL = "proxyUrl";
+    public static final String PROXY_PORT = "proxyPort";
+    public static final String INSIGHTS_ENDPOINT = "insightsEndpoint";
+    public static final String SYSTEM_ENDPOINT = "systemEndpoint";
+    public static final String USER_AGENT = "userAgent";
+
+    public static final String JBOSS_PROPERTY_DIR = "jboss.server.data.dir";
+
+    public static final String INSIGHTS_PROPERTY_FILE_NAME = "insights.properties";
+
+    public static final String INSIGHTS_DESCRIPTION = "Properties file consisting of RHN login information and insights/insights URL";
+
+    public static final int DEFAULT_SCHEDULE_INTERVAL = 1;
+    public static final String DEFAULT_BASE_URL = "https://api.access.redhat.com";
+    public static final String DEFAULT_INSIGHTS_ENDPOINT = "/r/insights/v1/uploads/";
+    public static final String DEFAULT_SYSTEM_ENDPOINT = "/r/insights/v1/systems/";
+    public static final String DEFAULT_USER_AGENT = "redhat-support-lib-java";
+
+    private InsightsJdrScheduler scheduler;
+    private boolean enabled = true;
+    private ScheduledExecutorService executor;
+
+    private InjectedValue<JdrReportCollector> jdrCollector = new InjectedValue<>();
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public synchronized void start(StartContext context) throws StartException {
+        try {
+            if (enabled) {
+                scheduler.startScheduler();
+            }
+        } catch (IllegalStateException e) {
+            throw new StartException(e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public synchronized void stop(StopContext context) {
+        final InsightsJdrScheduler scheduler = this.scheduler;
+        this.scheduler = null;
```

Comment:
Since this is nulled out here, you must construct a new one in start(). It's a requirement that a Service can be started and stopped multiple times. So the general pattern is to construct in start() objects that will be nulled out in stop(), and don't construct them in the service impl's constructor (i.e. not in L105 below).